### PR TITLE
feat: move to the new scripting engine

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java [hunk 1]
```diff
@@ -4,27 +4,40 @@
...
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
 
 @SuperBuilder
 @ToString
 @EqualsAndHashCode
 @Getter
 @NoArgsConstructor
-public abstract class AbstractDbt extends AbstractBash implements RunnableTask<ScriptOutput> {
+public abstract class AbstractDbt extends AbstractExecScript implements RunnableTask<ScriptOutput> {
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:40]: I don't think the `extends AbstractExecScript` is needed, and even more it's exposed some props that would complexify the things (`interpreter`, `warningOnStdErr`)

File: src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java [hunk 2]
```diff
@@ -63,64 +76,139 @@ public abstract class AbstractDbt extends AbstractBash implements RunnableTask<S
...
-            if (this.projectDir != null) {
-                commands.add("--project-dir " + runContext.render(this.projectDir));
-            }
+    @Schema(
+        title = "Docker options for the `DOCKER` runner"
+    )
+    @PluginProperty
+    @Builder.Default
+    private DockerOptions docker = DockerOptions.builder()
+        .image("ghcr.io/dbt-labs/dbt-core")
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: @anna-geller would prefer our docker image, (that would also remove the complexity of entrypoint)

**Review by tchiotludo — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java [hunk 3]
```diff
@@ -63,64 +76,139 @@ public abstract class AbstractDbt extends AbstractBash implements RunnableTask<S
...
+        .entryPoint(List.of())
+        .build();
 
-            if (this.debug) {
-                commands.add("--debug");
-            }
+    @Schema(title = "Deprecated, use the `docker` property instead", deprecated = true)
+    @PluginProperty
+    @Deprecated
+    public DockerOptions getDockerOptions() {
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:120]: without the getter the property is not added to the JSONSchema so old flow will no longer validate.

File: src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java (hunk 2 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: I thought we use the standard by default even if in example we use our own images;
As I understand it, for dbt, you should use an image with dependencies installed, so by default this should be an image with no dependencies as we don't know which one is needed that's why I uses dbt-core

Comment (anna-geller) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: we can use DuckDB by default - it's super lightweight and doesn't require any external service: "ghcr.io/kestra-io/dbt-duckdb:latest"

Comment (anna-geller) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: `hcr.io/kestra-io/dbt-duckdb:latest`

as in this blueprint:

```yaml
id: dbtGitDockerDuckDB
namespace: blueprint

tasks:
  - id: dbt
    type: io.kestra.core.tasks.flows.WorkingDirectory
    tasks:
    - id: cloneRepository
      type: io.kestra.plugin.git.Clone
      url: https://github.com/kestra-io/dbt-demo
      branch: main

    - id: dbt-build
      type: io.kestra.plugin.dbt.cli.DbtCLI
      runner: DOCKER
      docker:
        image: ghcr.io/kestra-io/dbt-duckdb:latest
      profiles: |
        jaffle_shop:
          outputs:
            dev:
              type: duckdb
              path: ":memory:"
              extensions: 
                - parquet
              fixed_retries: 1
              threads: 16
              timeout_seconds: 300
          target: dev
      commands:
        - dbt deps
        - dbt build
```
can even be added to unit tests

File: src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tchiotludo) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:120]: it's a good way to force the migration no?

File: src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java (hunk 2 above)
Comment (anna-geller) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: dbt core is not useful for anything by itself without adapter and people often use the DuckDB dbt adapter for local/testing environments across all data warehouses (Snowflake, Reshift, BigQuery, etc) so it makes sense to use it as the default image

Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: I think we should not put any default as the user must select the image it needs

Comment (anna-geller) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: fair, the example already uses one of our dbt images (`ghcr.io/kestra-io/dbt-bigquery:latest`) so can be left empty - LGTM! 👍

Comment (anna-geller) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: good idea, let me check if Python will allow that :) could be dependency conflicts when trying to install all adapters in one package

Comment (anna-geller) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java:113]: several dependency conflicts, you can see the logs https://github.com/kestra-io/examples/actions/runs/5871922953/job/15922419233 but the build succeeded and I'm able to import all the libraries:
image

so it should be OK to use `ghcr.io/kestra-io/dbt:latest` as the default image 👍

Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/dbt/cli/AbstractDbt.java]: Done

**Pull request merged**